include/internal/bio_tfo.h

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2022 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/*
+ * Contains definitions for simplifying the use of TCP Fast Open
+ * (RFC7413) in OpenSSL socket BIOs.
+ */
+
+/* If a supported OS is added here, update test/bio_tfo_test.c */
+#if defined(TCP_FASTOPEN) && !defined(OPENSSL_NO_TFO)
+
+# if defined(OPENSSL_SYS_MACOSX) || defined(__FreeBSD__)
+#  include <sys/sysctl.h>
+# endif
+
+/*
+ * OSSL_TFO_SYSCTL is used to determine if TFO is supported by
+ * this kernel, and if supported, if it is enabled. This is more of
+ * a problem on FreeBSD 10.3 ~ 11.4, where TCP_FASTOPEN was defined,
+ * but not enabled by default in the kernel, and only for the server.
+ * Linux does not have sysctlbyname(), and the closest equivalent
+ * is to go into the /proc filesystem, but I'm not sure it's
+ * worthwhile.
+ *
+ * On MacOS and Linux:
+ * These operating systems use a single parameter to control TFO.
+ * The OSSL_TFO_SERVER_FLAG and OSSL_TFO_CLIENT_FLAGS are used to
+ * determine if TFO is enabled for the server and client respectively.
+ *
+ * OSSL_TFO_SERVER_FLAG = 1 = client TFO enabled
+ * OSSL_TFO_CLIENT_FLAG = 2 = server TFO enabled
```

Comment:
Errr....OSSL_TFO_CLIENT_FLAG indicates the *server* is enabled???

Reply:
Copying error. Thanks. It get's complicated because on Linux/Mac two bits of a single field are used to indicate enablement, but on FreeBSD, there are separate fields that indicate enablement. The flag values are correct, the comment following is wrong.